src/app/shared/services/loading.service.ts

```diff
@@ -0,0 +1,40 @@
+import { Overlay, OverlayRef } from '@angular/cdk/overlay';
+import { ComponentPortal } from '@angular/cdk/portal';
+import { Injectable } from '@angular/core';
+import { defer, NEVER } from 'rxjs';
+import { finalize, share } from 'rxjs/operators';
+import { LoadingComponent } from '../components/loading/loading.component';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class LoadingService {
+  private overlayRef: OverlayRef | undefined;
+
+  constructor(private readonly overlay: Overlay) {}
+
+  public readonly spinner$ = defer(() => {
+    this.show();
+    return NEVER.pipe(
+      finalize(() => {
+        this.hide();
+      })
+    );
+  }).pipe(share());
+
+  private show(): void {
+    // Hack avoiding `ExpressionChangedAfterItHasBeenCheckedError` error
+    Promise.resolve(null).then(() => {
```

Comment:
You don't need to pass a value to `Promise.resolve` if you don't use it:
```suggestion
    Promise.resolve().then(() => {
```

Reply:
poprawione